src/test/regress/expected/multi_colocation_utils.out

```diff
@@ -339,3 +339,263 @@ SELECT find_shard_interval_index(1300016);
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            1 |          2 |                 2 |                     23
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
+            4 |          4 |                 2 |                     23
+(4 rows)
+
+SELECT logicalrelid, colocationid FROM pg_dist_partition
+    WHERE colocationid >= 1 AND colocationid < 1000 
+    ORDER BY colocationid;
+ logicalrelid  | colocationid 
+---------------+--------------
+ table1_groupa |            1
+ table2_groupa |            1
+ table1_groupb |            2
+ table2_groupb |            2
+ table1_groupc |            3
+ table2_groupc |            3
+ table1_groupd |            4
+ table2_groupd |            4
+ table3_groupd |            4
+(9 rows)
+
+-- check effects of dropping tables
+DROP TABLE table1_groupA;
+SELECT * FROM pg_dist_colocation WHERE colocationid = 1;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            1 |          2 |                 2 |                     23
+(1 row)
+
+-- dropping all tables in a colocation group also deletes the colocation group
+DROP TABLE table2_groupA;
+SELECT * FROM pg_dist_colocation WHERE colocationid = 1;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+(0 rows)
+
+-- create dropped colocation group again
+SET citus.shard_count = 2;
+CREATE TABLE table1_groupE ( id int );
+SELECT create_distributed_table('table1_groupE', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupE ( id int );
+SELECT create_distributed_table('table2_groupE', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- check metadata
+SELECT * FROM pg_dist_colocation 
+    WHERE colocationid >= 1 AND colocationid < 1000 
+    ORDER BY colocationid;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
+            4 |          4 |                 2 |                     23
+            5 |          2 |                 2 |                     23
+(4 rows)
+
+-- cross check with internal colocation API
+SELECT 
+    p1.logicalrelid::regclass AS table1,
+    p2.logicalrelid::regclass AS table2,
+    tables_colocated(p1.logicalrelid , p2.logicalrelid) AS colocated
+FROM
+    pg_dist_partition p1,
+    pg_dist_partition p2
+WHERE 
+    p1.logicalrelid != p2.logicalrelid AND
+    p1.colocationid != 0 AND
+    p2.colocationid != 0 AND
```

Comment:
I guess if we add `p1.logicalrelid > p2.logicalrelid` we'll get rid of the repetitions